src/components/DataViewer/DataViewer.tsx

```diff
@@ -0,0 +1,111 @@
+"use client";
+
+import React, { ReactElement } from "react";
+import styled from "styled-components";
+import Modal from "react-modal";
+import CloseIcon from "@mui/icons-material/Close";
+
+interface DataViewerProps {
+    data: { [key: string]: string | number | null };
+    header: ReactElement | string;
+    isOpen: boolean;
+    onRequestClose: () => void;
+}
+
+const StyledModal = styled(Modal)`
+    display: flex;
+    flex-direction: column;
+    position: absolute;
+    top: 50%;
+    left: 50%;
+    transform: translate(-50%, -50%);
+    border: 0;
+    border-radius: 5px;
+    outline: none;
+    padding: 0;
+    width: 80%;
+    max-height: 80%;
+    background: rgb(255, 255, 255);
+    box-shadow: rgb(200, 200, 200) 0 2px 10px;
+    overflow: hidden;
+`;
+
+const Header = styled.h3`
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+    background-color: #eeeeee;
+    padding: 0.5rem;
+    margin-top: 0;
+`;
+
+const Content = styled.div`
+    overflow: scroll;
+    margin: 1.5rem;
+    margin-top: 1rem;
+`;
+
+const Key = styled.div`
+    color: grey;
+    font-size: small;
+`;
+
+const Value = styled.div`
+    font-size: medium;
+`;
+
+const EachItem = styled.div`
+    padding-bottom: 1rem;
+`;
+
+const CloseButton = styled.button`
+    border: 0;
+    background-color: red;
+    color: white;
+    display: flex;
+    justify-content: center;
+`;
+
+const ClearButton: React.FC<{ closeModal: () => void }> = (props) => {
+    return (
+        <CloseButton onClick={props.closeModal}>
+            <CloseIcon />
+        </CloseButton>
+    );
+};
+
+const DataViewer: React.FC<DataViewerProps> = (props) => {
+    const JSONContent = Object.entries(props.data).map(([key, value]) => {
+        return (
+            <EachItem key={key}>
+                <Key>{key.toUpperCase().replace("_", " ")}</Key>
+                <Value>{value ?? ""}</Value>
+            </EachItem>
+        );
+    });
+
+    const closeModal: () => void = () => {
+        props.onRequestClose();
+    };
+
+    if (!props.isOpen) {
+        return <></>;
+    }
+
+    return (
+        <StyledModal
+            isOpen={true}
+            onRequestClose={closeModal}
+            ariaHideApp={false}
```

Comment:
what's `ariaHideApp` do?

Reply:
"It is important for users of screenreaders that other page content be hidden (via the aria-hidden attribute) while the modal is open." 
https://reactcommunity.org/react-modal/accessibility/

I think it doesn't hurt to include the line in the code.

Reply:
I think it's great, I just didn't know what it did!
On that page, it looks like you should give it the root element rather than 'false':

> To allow react-modal to do this, you should call Modal.setAppElement with a query selector identifying the root of your app

```
Modal.setAppElement('#root');
```

> If you are already applying the aria-hidden attribute to your app content through other means, you can pass the ariaHideApp={false} prop to your modal to avoid getting a warning that your app element is not specified.

so I think with this you are actually turning off the warning?

Reply:
You are completely correct. We have now added the setAppElement and set ariaHideApp to true. Thanks.